controllers/nnf_node_storage_controller.go

```diff
@@ -584,6 +601,49 @@ func (r *NnfNodeStorageReconciler) formatFileSystem(ctx context.Context, nodeSto
 	return nil, nil
 }
 
+func (r *NnfNodeStorageReconciler) setLustreOwnerGroup(nodeStorage *nnfv1alpha1.NnfNodeStorage) error {
+	log := r.Log.WithValues("NnfNodeStorage", types.NamespacedName{Name: nodeStorage.Name, Namespace: nodeStorage.Namespace})
+
+	_, found := os.LookupEnv("NNF_TEST_ENVIRONMENT")
+	if found || os.Getenv("ENVIRONMENT") == "kind" {
+		return nil
+	}
+
+	if nodeStorage.Spec.FileSystemType != "lustre" {
+		return fmt.Errorf("Invalid file system type '%s' for setting owner/group", nodeStorage.Spec.FileSystemType)
+	}
+
+	target := "/mnt/nnf/client/" + nodeStorage.Name
+	if err := os.MkdirAll(target, 0755); err != nil {
+		log.Error(err, "Mkdir failed")
+		return err
+	}
+	defer os.RemoveAll(target)
+
+	mounter := mount.New("")
+	mounted, err := mounter.IsMountPoint(target)
+	if err != nil {
+		return err
+	}
+
+	source := nodeStorage.Spec.LustreStorage.MgsNode + ":/" + nodeStorage.Spec.LustreStorage.FileSystemName
+
+	if !mounted {
+		if err := mounter.Mount(source, target, "lustre", nil); err != nil {
+			log.Error(err, "Mount failed")
+			return err
+		}
+	}
+	defer func() { err = mounter.Unmount(target) }()
```

Comment:
You don't have 'err' as a named return value, and if you did it would overwrite an error from os.Chown below.

Reply:
Oops, I forgot to add the fix here. I'll fix it.